homeassistant/components/azure_event_hub/config_flow.py

```diff
@@ -0,0 +1,200 @@
+"""Config flow for azure_event_hub integration."""
+from __future__ import annotations
+
+from copy import deepcopy
+import logging
+from typing import Any
+
+from azure.eventhub.exceptions import EventHubError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .client import AzureEventHubClient
+from .const import (
+    CONF_EVENT_HUB_CON_STRING,
+    CONF_EVENT_HUB_INSTANCE_NAME,
+    CONF_EVENT_HUB_NAMESPACE,
+    CONF_EVENT_HUB_SAS_KEY,
+    CONF_EVENT_HUB_SAS_POLICY,
+    CONF_MAX_DELAY,
+    CONF_SEND_INTERVAL,
+    CONF_USE_CONN_STRING,
+    DEFAULT_OPTIONS,
+    DOMAIN,
+    STEP_CONN_STRING,
+    STEP_SAS,
+    STEP_USER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+BASE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EVENT_HUB_INSTANCE_NAME): str,
```

Comment:
Can we remove this required input? We don't seem to use it for anything besides the title and description placeholder. We can use a default name for title.

Reply:
No, we can't it is the only universal input into the connection, needed for both SAS and connection string methods, see l48 and l66 in client.py!